src/TrackGenerator.cpp

```diff
@@ -559,7 +605,9 @@ void TrackGenerator::generateTracks() {
     }
   }
 
+  /* Initialize the track boundary conditions and set the track UIDs */
   initializeBoundaryConditions();
+  initializeTrackPeriodicCycleIndices();
```

Comment:
In 2D and 3D, we only need the periodic track indices if periodic BCs are present. In 3D, we always need reflective cycle indices irregardless of the BCs. Currently, in 3D-MOC, there is not a separate routine specifically for creating the reflective cycle indices. The reflective cycle indices for 3D tracks are set in the routine where the 3D tracks are created.

I like the idea of a single routine called something line `initializeTrackCycleIndices(boundaryType bc)` that accepts a `boundaryType` enum with the type of cycle to set indices for. Then, the track will have both `_periodic_track_index` and `_reflective_track_index` even though we don't need the `_reflective_track_index` at the moment. What do you guys think of that?